client/coral-embed/index.js

```diff
@@ -0,0 +1,163 @@
+  /* eslint-disable */
+  if (typeof define === 'function' && define.amd) {
+
+    // AMD. Register as an anonymous module.
+    define([], factory);
+  } else if (typeof module === 'object' && module.exports) {
+
+    // Node. Does not work with strict CommonJS, but
+    // only CommonJS-like environments that support module.exports,
+    // like Node.
+    module.exports = factory();
+  } else {
+
+    // Browser globals (root is window)
+    root.Coral = factory();
+  }
+  /* eslint-enable */
+}(this, function () {
+
+  // This function should return value of window.Coral
+  let pym = requirePym();
+  let Coral = {};
+  let Talk = Coral.Talk = {};
+
+  /**
+   * Render a Talk stream
+   * @param {HTMLElement} el - Element to render the stream in
+   * @param {Object} opts - Configuration options for talk
+   * @param {String} opts.talk - Talk base URL
+   * @param {String} [opts.title] - Title of Stream (rendered in iframe)
+   * @param {String} [opts.asset] - parent Asset ID or URL. Comments in the
+   * stream will replies to this asset
+   */
+  Talk.render = function (el, opts) {
+    if ( ! el) {
+      throw new Error('Please provide Coral.Talk.render() the HTMLElement you want to render Talk in.');
+    }
+    if (typeof el !== 'object') {
+      throw new Error(`Coral.Talk.render() expected HTMLElement but got ${  el  } (${  typeof el})`);
+    }
+    opts = opts || {};
+
+    // @todo infer this URL without explicit user input (if possible, may have to be added at build/render time of this script)
+    if (! opts.talk) {
+      throw new Error('Coral.Talk.render() expects opts.talk as the Talk Base URL');
+    }
+
+    // ensure el has an id, as pym can't directly accept the HTMLElement
+    if ( ! el.id) {el.id = `_${  String(Math.random())}`;}
+    let asset = opts.asset || window.location;
+    let pymParent = new pym.Parent(
+      el.id,
+      buildStreamIframeUrl(opts.talk, asset),
+      {
+        title: opts.title,
+        asset_url: asset,
+        id: `${el.id  }_` + 'iframe',
+        name: `${el.id  }_` + 'iframe'
+      }
+    );
+
+    configurePymParent(pymParent, asset);
+  };
+
+  return Coral;
+
+  // build the URL to load in the pym iframe
+  function buildStreamIframeUrl(talkBaseUrl, asset) {
+    let iframeUrl = [
+      talkBaseUrl,
+      (talkBaseUrl.match(/\/$/) ? '' : '/'), // make sure no double-'/' if opts.talk already ends with '/'
+      'embed/stream?asset_url=',
+      encodeURIComponent(asset)
+    ].join('');
+    return iframeUrl;
+  }
+
+  // Set up postMessage listeners/handlers on the pymParent
+  // e.g. to resize the iframe, and navigate the host page
+  function configurePymParent(pymParent, assetUrl) {
+    let notificationOffset = 200;
+    let ready = false;
+
+    // Resize parent iframe height when child height changes
+    pymParent.onMessage('height', function(height) {
+
+      // TODO: In local testing, this is firing nonstop. Maybe there's a bug on the inside?
```

Comment:
Heads up, I noticed this while devving and may be worth creating a task in pivotal to investigate. Basically the 'height' callback is just constantly executing when talk is on a page. And because it updates the DOM each time it's going to have a big negative effect on page performance.